### PR TITLE
feat : added dynamic cart bill

### PR DESCRIPTION
**Pull request by JuvitaSaini09**
### Added Cart 

Files to be checked :

- cartItems.js
- priceTable.js
- cartContext.js
- cartPage.css
- cartPage.js

**Snapshot of cart page**
image

### CODE REVIEW

File: src/components/productListing/productListing.js [hunk 1]
```diff
@@ -32,7 +34,7 @@ function ProductListing() {
                     </div>
                   </a>
                   <footer>
-                    <button className="btn-cart" >Add to cart</button>
+                    <button className="btn-cart" onClick={()=>{dispatch({type:"ADD_TO_CART",book:item})}}>Add to cart</button>
```
Comment (gauravsinhaweb) [inline src/components/productListing/productListing.js:37]: make separate handler for this later

**Review by gauravsinhaweb — APPROVED** [review]
Good to merge!

File: src/components/cart/priceTable.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    const countOfItems = arrayOfCountOfItemsInCart.reduce((previousValue, currentValue) => previousValue + currentValue, 0)
+
+    //update bill
+    setBill({ priceOfItems: sumOfItemsInCart, deliveryCharge: 50, totalPrice: 50 + sumOfItemsInCart, count: countOfItems })
+  }, [state]);
+  console.log(bill)
+
+  return (
+    <>
+      {state[0] === undefined ? "" : <section className="col-2">
```
Comment (yaswanthmyneni) [inline src/components/cart/priceTable.js]: suggestion: check length of state

File: src/components/cart/priceTable.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    const arrayOfCountOfItemsInCart = state.map(item => {
+      return Number(item.count)
+    })
+    //count of all items in cart
+    const countOfItems = arrayOfCountOfItemsInCart.reduce((previousValue, currentValue) => previousValue + currentValue, 0)
+
+    //update bill
+    setBill({ priceOfItems: sumOfItemsInCart, deliveryCharge: 50, totalPrice: 50 + sumOfItemsInCart, count: countOfItems })
+  }, [state]);
+  console.log(bill)
```
Comment (yaswanthmyneni) [inline src/components/cart/priceTable.js]: changes: remove console log when you push code to github

**Review by yaswanthmyneni — APPROVED** [review]
you can merge it

File: src/components/cart/priceTable.js (hunk 3 above)
Comment (JuvitaSaini09) [reply src/components/cart/priceTable.js]: ok,i have removed it thanks for the pr review

**Pull request merged**